### PR TITLE
scripts/compileFsharpScripts.fsx: add

### PR DESCRIPTION
**Pull request by realmarv**

### CODE REVIEW

**Comment by knocte** [comment]
2 things left to do here:

- Fix CI.
- Add an invocation to this script, in CI.

**Comment by knocte** [comment]
This PR should update the ReadMe too.

File: scripts/compileFsharpScripts.fsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        Fsdk.Process.Echo.All
+    )
+    .UnwrapDefault()
+|> ignore<string>
+
+let rootDir = Path.Combine(__SOURCE_DIRECTORY__, "..") |> DirectoryInfo
+
+Directory.GetFiles(rootDir.FullName, "*.fsx", SearchOption.AllDirectories)
+|> Seq.iter(
+    function
```
Comment (knocte) [inline scripts/compileFsharpScripts.fsx:37]: @realmarv using `fun` is shorter

Comment (realmarv) [reply scripts/compileFsharpScripts.fsx]: Done

**Comment by realmarv** [comment]
> 2 things left to do here:
> 
> * Fix CI.
> * Add an invocation to this script, in CI.

Done

**Comment by realmarv** [comment]
> This PR should update the ReadMe too.

Done

**Comment by knocte** [comment]
This PR needs to be rebased unfortunately.

**Comment by realmarv** [comment]
> This PR needs to be rebased unfortunately.

Done

File: ReadMe.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ This is a repository that contains several useful things that other `nblockchain
         * [Use of `-latest` suffix in `runs-on:` GitHubCI tags](scripts/unpinnedGitHubActionsImageVersions.fsx).
         * [Use of asterisk (*) in `PackageReference` items of .NET projects](scripts/unpinnedDotnetPackageVersions.fsx).
         * [Missing the version number in `#r "nuget: ` refs of F# scripts](scripts/unpinnedNugetPackageReferenceVersions.fsx).
+        * [Compile F# scripts](scripts/compileFsharpScripts.fsx).
```
Comment (knocte) [inline ReadMe.md:15]: @realmarv you placed this as an item part of a list named "use of unpinned versions". This is not about versions. Please pay more attention to detail.

File: scripts/compileFsharpScripts.fsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        },
+        Fsdk.Process.Echo.All
+    )
+    .UnwrapDefault()
+|> ignore<string>
+
+let rootDir = Path.Combine(__SOURCE_DIRECTORY__, "..") |> DirectoryInfo
+
+Helpers.GetFiles rootDir "*.fsx"
+|> Seq.map(fun fileInfo -> fileInfo.FullName)
```
Comment (knocte) [inline scripts/compileFsharpScripts.fsx:37]: @realmarv there is no need to do this here, just use the .FullName property inside the iter lambda

**Comment by knocte** [comment]
Why is CI red?

**Pull request merged**